### PR TITLE
Fix `destroyClient` is not a function in Wasm

### LINKED ISSUE #871: [Bug]: `destroyClient` is not a function
### Issue description

We're getting a Typescript error and runtime error related to the `destroyClient` function. The TS error, when importing `sdk-wasm` is:

```
node_modules/@iota/sdk-wasm/node/lib/bindings.d.ts:2:36 - error TS2305: Module '"../wasm/iota_sdk_wasm"' has no exported member 'destroyClient'.

2 import { initLogger, createClient, destroyClient, createSecretManager, createWallet, callClientMethodAsync, callSecretManagerMethodAsync, callWalletMethodAsync, destroyWallet, listenWalletAsync, getClientFromWallet, getSecretManagerFromWallet, listenMqtt, migrateStrongholdSnapshotV2ToV3 } from '../wasm/iota_sdk_wasm';
                                     ~~~~~~~~~~~~~

Found 1 error in node_modules/@iota/sdk-wasm/node/lib/bindings.d.ts:2
```

### Version

`@iota/sdk-wasm@1.0.0`

### Expected behaviour

The reproduction example succeeds.

### Actual behaviour

The example fails with `TypeError: (0 , bindings_1.destroyClient) is not a function`

### Can the issue reliably be reproduced?

Yes

### Steps to reproduce the issue

1. Add the `Client` import to `bindings/wasm/examples/node.js`.
2. In the example, call `new Client({}).destroy();`. 
3. Execute.

### Errors

```
~/git/iota-sdk/bindings/wasm/node/lib/client/client-method-handler.js:19
        return (0, bindings_1.destroyClient)(this.methodHandler);
                                            ^

TypeError: (0 , bindings_1.destroyClient) is not a function
    at ClientMethodHandler.destroy (/home/philipp/git/iota-sdk/bindings/wasm/node/lib/client/client-method-handler.js:19:45)
    at Client.destroy (/home/philipp/git/iota-sdk/bindings/wasm/node/lib/client/client.js:18:35)
    at run (/home/philipp/git/iota-sdk/bindings/wasm/examples/node.js:45:20)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

Node.js v18.12.1
```

### Duplicate declaration

- [X] I have searched the issues tracker this issue and there is none

### PR DESCRIPTION
**Pull request by abdulmth**
## Links to any relevant issues

fixes #871

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

WASM build works.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by Thoralf-M — DISMISSED** [review]

File: bindings/wasm/package-lock.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
 {
     "name": "@iota/sdk-wasm",
-    "version": "1.0.0-rc.0",
+    "version": "1.0.0",
```
Comment (thibault-martinez) [inline bindings/wasm/package-lock.json]: Update it again?

File: bindings/wasm/CHANGELOG.md [hunk 2]
```diff
@@ -19,6 +19,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Security -->
 
+## 1.0.1 - 2023-07-25
+
+### Fixed
+
+- `destroyClient` is not a function.
```
Comment (thibault-martinez) [inline bindings/wasm/CHANGELOG.md]:
```suggestion
- `destroyClient` is not a function;
```

File: bindings/wasm/src/client.rs [hunk 3]
```diff
@@ -37,6 +37,12 @@ pub fn create_client(clientOptions: String) -> Result<ClientMethodHandler, JsVal
     Ok(ClientMethodHandler { client })
 }
 
+/// Necessary for compatibility with the node.js bindings.
+#[wasm_bindgen(js_name = destroyClient)]
+pub async fn destroy_client(_client_method_handler: &ClientMethodHandler) -> Result<(), JsValue> {
```
Comment (thibault-martinez) [inline bindings/wasm/src/client.rs:42]: So for my understanding, what does this mean exactly? We do not have an implementation for this in Wasm?

File: bindings/wasm/CHANGELOG.md [hunk 4]
```diff
@@ -19,6 +19,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Security -->
 
+## 1.0.1 - 2023-07-25
+
+### Fixed
+
+- `destroyClient` is not a function;
```
Comment (thibault-martinez) [inline bindings/wasm/CHANGELOG.md]: This feels a bit weird in a changelog tbh, can you think of a more descriptive sentence for what was actually fixed?

File: bindings/wasm/src/client.rs (hunk 3 above)
Comment (abdulmth) [reply bindings/wasm/src/client.rs:42]: Yes, in node it's under an `Arc`, but in WASM it's not, so it doesn't require a destroy method.

Comment (thibault-martinez) [reply bindings/wasm/src/client.rs:42]: Thanks

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**